portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -43,14 +44,16 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     // Prepares results and sorts them in descending order.
     Query query = new Query("entry").addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
+    // TODO(aabundis): Find efficient way other than clearing arrayList and then filling up again.
     comments.clear();
-    for (Entity entity : results.asIterable()) {
+    int limit = toInt(request.getParameter("max"));
+    for (Entity entity : results.asList(FetchOptions.Builder.withLimit(limit))) {
       String comment = (String) entity.getProperty("com");
       String firstN = (String) entity.getProperty("firstN");
```

Comment:
Consider changing this to 'firstName' and 'lastName', because that's much more clear than 'firstN' to people reading your code!

Reply:
Done!

Reply:
Ack!